caravel/assets/javascripts/SqlLab/components/QuerySearch.jsx

```diff
@@ -34,13 +36,41 @@ class QuerySearch extends React.Component {
     const val = (db) ? db.value : null;
     this.setState({ databaseId: val });
   }
-  insertParams(baseUrl, params) {
-    return baseUrl + '?' + params.join('&');
+  getTimeFromSelection(selection) {
+    switch (selection) {
```

Comment:
i think switch/case works fine in this case. i think the function names could be made more clear. what is the advantage of using an object mapping over case/switch here @bkyryliuk ?